tests/unit/rpc/BaseTests.cpp

```diff
@@ -353,6 +353,39 @@ TEST_F(RPCBaseTest, WithCustomError)
     ASSERT_EQ(err.error(), ripple::rpcALREADY_MULTISIG);
 }
 
+TEST_F(RPCBaseTest, TimeFormatValidator)
+{
+    auto const spec = RpcSpec{
+        {"date", TimeFormatValidator{"%Y-%m-%dT%H:%M:%SZ"}},
+    };
+
+    auto passingInput = json::parse(R"({ "date": "2023-01-01T00:00:00Z" })");
+    ASSERT_TRUE(spec.process(passingInput));
```

Comment:
Keep in mind that unlike EXPECT_\*, ASSERT_\* immediately terminates the test and you only get the first error out of all. Sometimes it's desired but sometimes not